[SwiftArray] Pass an execution scope when getting the size of the array.

dcci opened the pull request:
For non-fixed layout types, we have to ask the runtime what's the
size of the type (and go through mirrors). This shows up with resilient
types, but probably it was a more general problem.
Thanks to Adrian for help to initially diagnose this!

<rdar://problem/51146107>

dcci commented:
@swift-ci test

adrian-prantl reviewed (commented):
Line SwiftArray.cpp:197 seems to have the exact same bug.

Commit:
[SwiftArray] Pass an execution scope when getting the size of the array.

For non-fixed layout types, we have to ask the runtime what's the
size of the type (and go through mirrors). This shows up with resilient
types, but probably it was a more general problem.
Thanks to Adrian for help to initially diagnose this!

<rdar://problem/51146107>

dcci commented:
@adrian-prantl thanks for the catch. Updated.

dcci commented:
I'll audit all the `getBitSize()` next, but this should be good for now.

dcci commented:
@swift-ci test

adrian-prantl commented:
Great! GetBitStride is also a likely candidate, if we use it at all.

Commit:
[test] This one runs only on macOS.

dcci commented:
@swift-ci test and merge
